apps/client/src/features/table/table-settings/TableSettings.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, Checkbox, Switch } from '@chakra-ui/react';
+import { Column } from '@tanstack/react-table';
+import { OntimeRundownEntry } from 'ontime-types';
+
+import style from './TableSettings.module.scss';
+
+// reusable button styles
+const buttonProps = {
+  size: 'sm',
+  variant: 'ontime-subtle',
+};
+
+interface CuesheetSettingsProps {
+  columns: Column<OntimeRundownEntry, unknown>[];
+  handleResetResizing: () => void;
+  handleResetReordering: () => void;
+  handleClearToggles: () => void;
+}
+
+export function CuesheetSettings(props: CuesheetSettingsProps) {
+  const { columns, handleResetResizing, handleResetReordering, handleClearToggles } = props;
+
+  return (
+    <div className={style.tableSettings}>
+      <div className={style.leftPanel}>
+        <div className={style.sectionTitle}>Toggle field visibility</div>
+        <div className={style.options}>
+          {columns.map((column) => {
+            const columnHeader = column.columnDef.header;
+            const visible = column.getIsVisible();
+
+            return (
+              <label key={column.id} className={style.option}>
```

Comment:
Seems like toggling all the columns doesn't re-enable the checkboxes themselves. Adding `visible` to the `key` seems to do the trick:

```suggestion
              <label key={`${column.id}${visible}`} className={style.option}>
```